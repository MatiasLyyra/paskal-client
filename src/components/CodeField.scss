@import "src/assets/color";

.CodeField-code {
  font-size: 3rem;
  width: calc(100% - 1rem);
  min-height: 10rem;
  background-color: $black;
  border: 2px solid $dark-green;
  color: $terminal-green;
  padding: 0.5rem;
}