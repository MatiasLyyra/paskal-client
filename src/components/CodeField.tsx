import React from "react";
import "./CodeField.scss";

export type IProps = {};
export interface IState {
  code: string;
  rawCode: string;
}

export class CodeField extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      code: "",
      rawCode: ""
    };
  }
  render() {
    return (
      <div>
        <div
          ref="rawCode"
          contentEditable={true}
          className="CodeField-code"
          onInput={this.updateCode.bind(this)}
        ></div>
        <textarea
          className="CodeField-Code"
          readOnly
          value={this.state.code}
        ></textarea>
      </div>
    );
  }
  updateCode() {
    const eventElemnt = this.refs.rawCode as HTMLElement;
    this.setState({
      code: this.removeHtmlTags(eventElemnt.innerHTML)
    });
  }
  removeHtmlTags(code: string): string {
    return code.replace(/<\/?div>/g, "\n").replace(/<br>/g, "");
  }
}

export default CodeField;
