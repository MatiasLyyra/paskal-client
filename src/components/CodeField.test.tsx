import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import CodeField from "./CodeField";

describe("CodeField text area", () => {
  let wrapper: ShallowWrapper;
  let instance: CodeField;
  let textArea: ShallowWrapper;
  let mockUpdate: jest.Mock;
  beforeEach(() => {
    mockUpdate = jest.fn();
    wrapper = shallow(<CodeField onCodeUpdate={mockUpdate} code="" indentation={2}></CodeField>);
    instance = wrapper.instance() as CodeField;
    textArea = wrapper.find("#rawCode");
  });
  it("updates code", () => {
    textArea.simulate("change", { target: { value: "foobar\n123" } });
    expect(mockUpdate).toHaveBeenLastCalledWith("foobar\n123")
    expect(wrapper).toMatchSnapshot();
  });
});
