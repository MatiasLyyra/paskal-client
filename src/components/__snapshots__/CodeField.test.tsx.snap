// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeField text area updates code 1`] = `
<div
  className="CodeField"
>
  <div
    className="CodeField-LineCount terminal-font"
  >
    <LineCount
      lines={1}
    />
  </div>
  <textarea
    autoCapitalize="false"
    autoComplete="false"
    autoCorrect="false"
    className="CodeField-code terminal-font"
    id="rawCode"
    onChange={[Function]}
    onKeyDown={[Function]}
    rows={5}
    spellCheck={false}
    value=""
  />
</div>
`;
